api/envoy/config/filter/http/jwt_authn/v2alpha/config.proto

```diff
@@ -183,44 +183,126 @@ message JwtHeader {
+  string issuer = 1;
+
+  // This audiences field overrides the one in JwtProvider if not empty.
+  repeated string audiences = 2;
+}
+
+// This message specifies a per route based Jwt config.
+// Route.per_filter_config for jwt_authn filter uses this message.
+message PerRouteConfig {
+  // Specify list of required JwtProviders. If list is empty,  Jwt is NOT required.
+  // If any of required JwtProviders is missing, or its verification fails, the request will
+  // be rejected.
+  // If a Jwt is not required, but it is in the providers list, it will be verified.
+  // But it will not fail the request if its verification fails.
+  repeated JwtRequirement requires = 1;
+}
+
+// This message specifies a Jwt requirement for a specific Route condition.
+// For example:
+//
+// .. code-block:: yaml
+//
+//    - match:
+//         prefix: "/healthz"
+//    - match:
+//         prefix: "/"
+//      - requires:
+//          issuer: jwt_issuer1
+//      - requires:
+//          issuer: jwt_issuer2
+//          audiences: [audience1, audience2]
+//
+message RequirementRule {
+  // The route matching parameter.  Only match is satisfied, the other fields will apply.
+  //
+  // For example: following match will match all requests.
+  //
+  // .. code-block:: yaml
+  //
+  //    match:
+  //       prefix: "/"
+  //
+  envoy.api.v2.route.RouteMatch match = 1 [(validate.rules).message.required = true];
+
+  // Specify list of required JwtProviders. If list is empty,  Jwt is NOT required.
+  // If any of required JwtProviders is missing, or its verification fails, the request will
+  // be rejected.
+  // If a Jwt is not required, but it is in the providers list, it will be verified.
+  // But it will not fail the request if its verification fails.
+  repeated JwtRequirement requires = 2;
+}
+
 // This is the Envoy HTTP filter config for JWT authentication.
-// [#not-implemented-hide:]
+//
+// For example:
+//
+// .. code-block:: yaml
+//
+//    - providers:
+//        issuer: issuer1
+//        audiences:
+//        - audience1
+//          audience2
+//        remote_jwks:
+//          http_uri:
+//            uri: https://example.com/.well-known/jwks.json
+//            cluster: example_jwks_cluster
+//    - providers:
+//        issuer: issuer2
+//        audiences:
+//        local_jwks:
+//          inline_string: jwks_string
+//    # Not jwt verification is required for /health path
+//    - rules:
+//        match:
+//          prefix: "/health"
+//    # Jwt verification for issuer1 is required for path prefixed with "prefix"
+//    - rules:
+//        match:
+//          prefix: "/prefix"
+//        - requires:
+//            issuer: issuer1
+//            audiences: [audience_override]
+//    # Jwt verification for issuer2 is required for all other requests.
+//    - rules:
+//        match:
+//          prefix: "/"
+//        - requires:
+//            issuer: issuer2
+//
+//// [#not-implemented-hide:]
```

Comment:
Same as above, check the YAML, I think "requires" fields are a little off.

Reply:
Done